sentry_sdk/integrations/asyncpg.py

```diff
@@ -60,3 +67,64 @@ async def _inner(*args: P.args, **kwargs: P.kwargs) -> T:
         return res
 
     return _inner
+
+
+def _wrap_basecursor_exec(f: Callable[P, T]) -> Callable[P, T]:
+    async def _exec(self: BaseCursor, n, timeout):
+        hub = Hub.current
+        integration = hub.get_integration(AsyncPGIntegration)
+
+        if integration is None:
+            return await f(self, n, timeout)
+
+        record_params = integration._record_params
+        params_list = self._args[1] if record_params else None
+        param_style = "pyformat" if params_list else None
+
+        executemany = n > 1
+
+        with record_sql_queries(
+            hub,
+            self,
+            self._query,
+            params_list,
+            param_style,
+            executemany=executemany,
+            record_cursor_repr=True,
+        ):
+            res = await f(self, n, timeout)
+        return res
+
+    return _exec
+
+
+def _wrap_cursoriterator_anext(f: Callable[P, T]) -> Callable[P, T]:
+    async def __await__(self: CursorIterator):
+        hub = Hub.current
+        integration = hub.get_integration(AsyncPGIntegration)
+
+        if integration is None:
+            return await f(self)
+
+        record_params = integration._record_params
+        params_list = self._args[1] if record_params else None
+        param_style = "pyformat" if params_list else None
+
+        with record_sql_queries(
+            hub,
+            self,
+            self._query,
+            params_list,
+            param_style,
+            executemany=False,
+            record_cursor_repr=True,
+        ) as span:
+            try:
+                res = await f(self)
+            except StopAsyncIteration:
+                span.set_data("db.cursor.exhausted", True)
```

Comment:
I'm really confused why this is not recorded. 

The code is reached, but it doesn't matter what data I'm setting on this span, it won't be recorded.
What am I missing here?

Reply:
Ok, I figured out, that we don't check for the data of the span in the test but for the data of the breadcrumb.

Is there a way to test for the data for a span?

Reply:
This is removed now, since we don't instrument every iteration of a cursor anymore.